backend/server.go

```diff
@@ -9,11 +9,12 @@ import (
 )
 
 type ZenaoServer struct {
-	Logger       *zap.Logger
-	GetUser      func(ctx context.Context) *zeni.AuthUser
-	CreateUser   func(ctx context.Context, email string) (*zeni.AuthUser, error)
-	Chain        zeni.Chain
-	DB           zeni.DB
-	MailClient   *resend.Client
-	DiscordToken string
+	Logger               *zap.Logger
+	GetUser              func(ctx context.Context) *zeni.AuthUser
+	GetUsersFromClerkIDs func(ctx context.Context, ids []string) ([]*zeni.AuthUser, error)
+	CreateUser           func(ctx context.Context, email string) (*zeni.AuthUser, error)
```

Comment:
maybe move these into a `zeni.Auth` interface and create a `ClerkAuth` implem

Reply:
agree :white_check_mark: , i start working on this on another PR to keep this one atomic